### DISCUSSION TITLE
Ferrum::Browser load inline page or from local file

### DISCUSSION THREAD

**Discussion by davidwessman** [rubycdp/ferrum#261] [General]
Hello.

I read through the documentation and the options for `Ferrum::Browser.new` but could not find any way to load a browser from a local file or string instead of an url.

The use-case I have is to render PDFs where there is no URL to load the HTML from.

```ruby
html = generate_html
browser = Ferrum::Browser.new
browser.load(html)
tmp = Tempfile.new("pdf-chrome-#{filename}")
browser.pdf(
  path: tmp.path,
  format: :A4,
  landscape: false,
  margin: {top: 36, right: 36, bottom: 36, left: 36},
  preferCSSPageSize: true,
  printBackground: true
)
browser.quit
pdf_data = File.read(tmp.path)
```

I am sorry if I missed some option for this, thankful for any help 🙂

**Comment by route** [comment]
You can workaround it with https://github.com/rubycdp/ferrum#interceptoptions and `request.respond(body: "Lorem ipsum")` in particular.

**Reply by dhnaranjo** [reply]
This suggestion worked great for me with pretty much the same problem.

**Comment by route** [comment]
In fact there's a better approach https://github.com/rubycdp/ferrum#content--html I forgot about

```ruby
browser = Ferrum::Browser.new
page = browser.create_page
page.content = "<html><head></head><body><p>lol</p></body></html>"
page.body # => <html><head></head><body><p>lol</p></body></html>
```

**Reply by davidwessman** [reply]
Oh, will try this out!

Currently running your previous suggestion in production and it is working well, but would be easier to avoid intercepting.

**Reply by davidwessman** [reply]
@route 
Got around to testing this today, but:

```ruby
browser = Ferrum::Browser.new
page = browser.create_page
page.content = html
NoMethodError: undefined method `content=' for #<Ferrum::Page:0x00007fee85195ea8>
```

**Reply by davidwessman** [reply]
Ah, it is `set_content` in the latest released version!

**Reply by route** [reply]
Yea try master, but I will release new version soon
